src/zenml/config/secrets_store_config.py

```diff
@@ -31,9 +33,51 @@ class SecretsStoreConfiguration(BaseModel):
 
     Attributes:
         type: The type of store backend.
+        class_path: The Python class path of the store backend. Should point to
+            a subclass of `BaseSecretsStore`. This is optional and only
+            required if the store backend is not one of the built-in
+            implementations.
     """
 
     type: SecretsStoreType
+    class_path: Optional[str] = None
+
+    @root_validator
+    def validate_external(cls, values: Dict[str, Any]) -> Dict[str, Any]:
```

Comment:
Just from a naming perspective, this should probably be `custom` instead of `external`? It switches between the two which is a little confusing